File: engine/src/com/magma/engine/utils/FieldsPad.java
package com.magma.engine.utils;

import com.badlogic.gdx.utils.reflect.ClassReflection;
import com.badlogic.gdx.utils.reflect.Field;
import com.badlogic.gdx.utils.reflect.ReflectionException;

public abstract class FieldsPad {
    public FieldsPad() {

    }

    public <T> void setValue(String name, T value) throws ReflectionException {
        Field field;
        try {
            field = ClassReflection.getField(getClass(), name);
            field.set(this, value);
        } catch (ReflectionException e) {
            throw e;
        }
    }

    @SuppressWarnings("unchecked")
    public <T> T getValue(String name) throws ReflectionException {
        Field field;
        try {
            field = ClassReflection.getField(getClass(), name);
            T value = (T) field.get(this); // On exception: Is field of different type?
            return value;
        } catch (ReflectionException e) {
            throw e;
        }
    }
}

File: engine/src/com/magma/engine/utils/FieldsPad.kt
package com.magma.engine.utils

import com.badlogic.gdx.utils.reflect.ClassReflection
import com.badlogic.gdx.utils.reflect.Field
import com.badlogic.gdx.utils.reflect.ReflectionException

abstract class FieldsPad {
    fun <T> setValue(name: String, value: T) {
        val field: Field
        try {
            field = ClassReflection.getField(javaClass, name)
            field[this] = value
        } catch (e: ReflectionException) {
            throw e
        }
    }

    fun <T> getValue(name: String): T {
        val field: Field
        return try {
            field = ClassReflection.getField(javaClass, name)
            field[this] as T
        } catch (e: ReflectionException) {
            throw e
        }
    }
}